pkg/vsphere/tasks/waiter.go

```diff
@@ -185,6 +175,43 @@ func IsRetryError(op trace.Operation, err error) bool {
 	}
 }
 
+// Add an additional retry error InvalidArgument
+// This is added for potential race conditions concerning vSAN host partition
+// Normally VM configuration is provided without user input. If there's InvalidArgument fault, it would more possibly be a system error.
+// Details see: https://github.com/vmware/vic/pull/4597
+func IsRetryError(op trace.Operation, err error) bool {
+	return IsTransientError(op, err) || IsInvalidArgumentError(err)
+}
+
+// IsInvalidArgumentError checks if a soap fault or vim fault is InvalidArgument error
+func IsInvalidArgumentError(err error) bool {
+	if soap.IsVimFault(err) {
+		if _, ok := soap.ToVimFault(err).(*types.InvalidArgument); ok {
```

Comment:
I wonder why do you use a pointer type and a type. For different type of faults. I looks a little inconsistent.
In this particular case I would probably write code a little different that is more friendly for the CPU branch prediction algorithms. Not saying it is really necessary, just saying that it triggers some of my memories.

```
_, ok1 := soap.ToVimFault(err).(*types.InvalidArgument)
_, ok2 := soap.ToVimFault(err).(types.InvalidArgumentFault)
return ok1 || ok2
```

Reply:
For some reason it is the only way that my IDE doesn't complain about type check...
If I use type instead of pointer type in line 189, it gave me errors since `ToVimFault` gives type `BaseMethodFault` and `InvalidArgument` does not implement that.

It does look a lot neater, thanks

Reply:
> that is more friendly for the CPU branch prediction algorithms
Can you elaborate on that?
I was thinking if the error could be caught in the first if clause, it doesn't have to execute second clause...I guess it's not necessarily efficient this way?

Reply:
if there is no IF and no data dependency, CPU most likely run both in parallel without you even knowing it.